### PR TITLE
Add missing getters

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin @sdroege

### CODE REVIEW

File: src/event_button.rs [hunk 1]
```diff
@@ -33,4 +33,14 @@ impl EventButton {
...
+
+    pub fn get_axes(&self) -> Option<(f64, f64)> {
+        let axes = self.as_ref().axes;
+
+        if axes.is_null() {
+            None
+        } else {
+            unsafe { Some((*axes, *axes.offset(1))) }
+        }
+    }
```
Comment (sdroege) [inline src/event_button.rs]: Please check the docs and add *all* the missing getters while we're at it :)

There's at least `x_root` and `y_root` still missing here and maybe other things for the other events

Comment (GuillaumeGomez) [reply src/event_button.rs]: Damn, I thought it was implemented in `Event`... I'll need to make a bigger check.

**Comment by GuillaumeGomez** [comment]
Added a *lot* of missing stuff.

File: src/lib.rs [hunk 2]
```diff
@@ -55,12 +55,20 @@ mod event_grab_broken;
...
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+mod event_pad_group_mode;
 mod event_property;
 mod event_proximity;
 mod event_scroll;
 mod event_selection;
 mod event_setting;
 mod event_touch;
+mod event_touchpad_pinch;
+mod event_touchpad_swipe;
```
Comment (sdroege) [inline src/lib.rs]: These two exist since 3.18

File: src/lib.rs [hunk 3]
```diff
@@ -55,12 +55,20 @@ mod event_grab_broken;
...
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+mod event_pad_button;
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+mod event_pad_group_mode;
 mod event_property;
 mod event_proximity;
 mod event_scroll;
 mod event_selection;
 mod event_setting;
 mod event_touch;
```
Comment (sdroege) [inline src/lib.rs]: `GDK_TOUCH_BEGIN/END/CANCEL/UPDATE` exist since 3.4, not sure how this worked without feature flag before?

File: src/lib.rs [hunk 4]
```diff
@@ -111,12 +119,20 @@ pub use event_grab_broken::EventGrabBroken;
...
+#[cfg(any(feature = "v3_22", feature = "dox"))]
+pub use event_pad_group_mode::EventPadGroupMode;
 pub use event_property::EventProperty;
 pub use event_proximity::EventProximity;
 pub use event_scroll::EventScroll;
 pub use event_selection::EventSelection;
 pub use event_setting::EventSetting;
 pub use event_touch::EventTouch;
+pub use event_touchpad_pinch::EventTouchpadPinch;
+pub use event_touchpad_swipe::EventTouchpadSwipe;
```
Comment (sdroege) [inline src/lib.rs]: 3.18

File: src/event_key.rs [hunk 5]
```diff
@@ -38,5 +38,8 @@ impl EventKey {
         self.as_ref().group
     }
 
-    // TODO: add getter for is_modifier
+    // TODO: to be uncommented when the field will be generated
+    // pub fn get_is_modifier(&self) -> bool {
+    //     from_glib(self.as_ref().is_modifier)
+    // }
```
Comment (sdroege) [inline src/event_key.rs]: Bitfields /o\

As the whole `GdkEvent` union is zero-allocated, you can
```
    let ptr = (&self.as_ref().group as *const u8).offset(1);
    from_glib(*ptr)
```

File: src/event_scroll.rs [hunk 6]
```diff
@@ -45,4 +45,9 @@ impl EventScroll {
         let dy = self.as_ref().delta_y;
         (dx, dy)
     }
+
+    // TODO: to be uncommented when the field will be generated
+    // pub fn get_is_stop(&self) -> bool {
+    //     from_glib(self.as_ref().is_stop)
```
Comment (sdroege) [inline src/event_scroll.rs]: See https://github.com/gtk-rs/gdk/pull/232/files#r203947133, same story

**Comment by sdroege** [comment]
Looks good

File: src/event_key.rs (hunk 5 above)
Comment (GuillaumeGomez) [reply src/event_key.rs]: I'd rather not. I prefer to just wait for the generation instead.

**Comment by GuillaumeGomez** [comment]
Then I merge!

File: src/event_key.rs (hunk 5 above)
Comment (sdroege) [reply src/event_key.rs]: We can probably wait years for that, it's far from trivial to get right for the general case

Comment (GuillaumeGomez) [reply src/event_key.rs]: I'm a patient man. :)

Comment (sdroege) [reply src/event_key.rs]: The users might not be :P

Comment (GuillaumeGomez) [reply src/event_key.rs]: I'll think about it when I'll hear their complains then. :p

**Pull request merged**